# Parking mandates

## Hypothesis

The off-street parking mandate may be requiring more space dedicated to parking than is necessary.

In [42]:
import pandas as pd
import requests

from data.connect_census import get_census_key

API_KEY = get_census_key()

## Fetch American Community Survey data for Waltham census tracts

In [43]:
YEAR = 2024

# ref. https://www2.census.gov/programs-surveys/decennial/2010/partners/pdf/FIPS_StateCounty_Code.pdf
STATE = "25"   # Massachusetts
COUNTY = "017"  # Middlesex

WALTHAM_TRACTS = {
    "368101", "368102", "368200", "368300", "368400",
    "368500", "368600", "368700", "368800", "368901",
    "368902", "369000", "369100",
}

def tract_label(fips: str) -> str:
    return f"{int(fips[:-2])}.{fips[-2:]}" if fips[-2:] != "00" else str(int(fips[:-2]))

dp04_vars = ["DP04_0057E", "DP04_0058E", "DP04_0059E", "DP04_0060E", "DP04_0061E"]

r = requests.get(
    f"https://api.census.gov/data/{YEAR}/acs/acs5/profile",
    params={"get": ",".join(dp04_vars), "for": "tract:*", "in": f"state:{STATE} county:{COUNTY}", "key": API_KEY},
)
r.raise_for_status()
rows = r.json()

data_profiles_df = (
    pd.DataFrame(rows[1:], columns=rows[0])
    .query("tract in @WALTHAM_TRACTS")
    .assign(tract=lambda df: df["tract"].map(tract_label))
    .set_index("tract")
    [dp04_vars]
    .apply(pd.to_numeric, errors="coerce")
    .sort_index()
)

data_profiles_df.head()

,DP04_0057E,DP04_0058E,DP04_0059E,DP04_0060E,DP04_0061E
tract,,,,,
3681.01,2380,268,798,1130,184
3681.02,1943,52,686,761,444
3682,1572,115,591,706,160
3683,2561,278,1149,903,231
3684,2384,108,1081,966,229


In [44]:
s0101_vars = ["S0101_C01_026E"]

r = requests.get(
    f"https://api.census.gov/data/{YEAR}/acs/acs5/subject",
    params={"get": ",".join(s0101_vars), "for": "tract:*", "in": f"state:{STATE} county:{COUNTY}", "key": API_KEY},
)
r.raise_for_status()
rows = r.json()

subject_df = (
    pd.DataFrame(rows[1:], columns=rows[0])
    .query("tract in @WALTHAM_TRACTS")
    .assign(tract=lambda df: df["tract"].map(tract_label))
    .set_index("tract")
    [s0101_vars]
    .apply(pd.to_numeric, errors="coerce")
    .sort_index()
)

subject_df.head()

,S0101_C01_026E
tract,
3681.01,4644
3681.02,4143
3682,3125
3683,4938
3684,8337


## Calculate number of adults per car

The usual assumption is that every adult in a household unit 'needs' a car, and any new construction should provide that amount of parking,
so incoming residents do not use on-street parking for vehicle storage.

This is set as ['2 per dwelling unit'](https://ecode360.com/26938091) for family dwellings.

This is assuming the typical 'nuclear' family with two adults who are doing the driving. If all adults are driving, one would expect that the number
of adults = the number of cars, so this ratio should come out to 1.

I have yet to find data that gets the number of cars in a census tract, so I have decided to calculate this from the survey questions as follows:

`n_cars = 1 * households with one car + 2 * households with two cars + 3 * households with 3 or more cars`

*Assumption* - in most cases, the number of cars in the '3 or more category' is going to tend to be closer to 3 than other values, for most homes.
If this assumption is wrong, `n_cars` will be lower than the actual amount.

In [45]:
# Estimate!!Total!!Total population!!SELECTED AGE CATEGORIES!!18 years and over
total_adults = subject_df["S0101_C01_026E"]

# # Estimate!!VEHICLES AVAILABLE!!Occupied housing units!!No vehicles available
# "DP04_0058E",
# # Estimate!!VEHICLES AVAILABLE!!Occupied housing units!!1 vehicle available
# "DP04_0059E",
# # Estimate!!VEHICLES AVAILABLE!!Occupied housing units!!2 vehicles available
# "DP04_0060E",
# # Estimate!!VEHICLES AVAILABLE!!Occupied housing units!!3 or more vehicles available
# "DP04_0061E"
parking_mandates_df = pd.DataFrame(columns=["total_vehicles", "total_adults", "adults_per_vehicle"])

parking_mandates_df["total_adults"] = subject_df["S0101_C01_026E"]


# ASSUMPTION - more than 3 vehicles will be rare, so assume 3 per household unit for that bucket
parking_mandates_df["total_vehicles"] = (1 * data_profiles_df["DP04_0059E"] + 2 * data_profiles_df["DP04_0060E"] + 3 * data_profiles_df["DP04_0061E"]).values

parking_mandates_df["adults_per_vehicle"] = parking_mandates_df["total_adults"] / parking_mandates_df["total_vehicles"]

print(parking_mandates_df)

         total_vehicles  total_adults  adults_per_vehicle
tract                                                    
3681.01            3610          4644            1.286427
3681.02            3540          4143            1.170339
3682               2483          3125            1.258558
3683               3648          4938            1.353618
3684               3700          8337            2.253243
3685               2273          3086            1.357677
3686               3999          4718            1.179795
3687               1621          2023            1.247995
3688               3769          4888            1.296896
3689.01            4933          5420            1.098723
3689.02            1847          2892            1.565782
3690                 54          3696           68.444444
3691               3718          4144            1.114578


## Comments

Most tracts come in at more than one adult per vehicle, except for tract 3690 and 3684.

3690 is mostly Bentley University, so we would expect many adults who are non-driving students.

3684 mostly overlaps ward 7, which is dominated by single family homes. It appears that many adults are sharing vehicles, but
this tract also contains Brandeis University. 

*Assumption* - It's not possible to tell if the all of the [18-24 year olds in tertiary eduation](https://data.census.gov/map?g=0400000US25_1400000US25017368101,25017368102,25017368200,25017368300,25017368400,25017368500,25017368600,25017368700,25017368800,25017368901,25017368902,25017369000,25017369100&tid=ACSST5Y2020.S1401&cid=S1401_C01_030E&layer=VT_2020_140_00_PY_D1&mode=thematic&loc=42.3818,-71.2223,z12.3362) in this tract are going to Brandeis, but assuming they are, and none of them are drivers, the ratio works out to be about the same as neighboring tracts (~1.15)

## Source

This data comes from the 5-year American Community Survey estimate from the YEAR value above. If anything, I would assume that the number
of vehicles would drop somewhat due to more adults no longer needing to commute to an office.

You can look at the uncombined data on the tracts [here](https://data.census.gov/map?q=DP04&g=1400000US25017368101,25017368102,25017368200,25017368300,25017368400,25017368500,25017368600,25017368700,25017368800,25017368901,25017368902,25017369000,25017369100&tid=ACSDP5Y2020.DP04&cid=DP04_0060PE&layer=VT_2020_140_00_PY_D1&mode=thematic&loc=42.3628,-71.2427,z13.6751). Change the `Variable(s)` field
to see how each tract fits in each category.